command/connect/envoy/envoy.go

```diff
@@ -299,9 +299,13 @@ func (c *cmd) run(args []string) int {
 			meta = map[string]string{structs.MetaWANFederationKey: "1"}
 		}
 
+		if c.proxyID == "" {
+			c.proxyID = c.gatewaySvcName
+		}
```

Comment:
I feel like this logic should probably live closer to the rest of the `proxyID` setup logic, potentially in another function. Otherwise I feel like it gets hard to reason about what is actually being set for `proxyID` depending on what flags you pass in.

Reply:
This is also technically changing the behavior of the CLI command, since previously, if `-proxy-id` was left empty, the command would attempt to lookup a gateway proxy from it's local agent.